### PR TITLE
Unit testing the 'browser' module.

### PR DESCRIPTION
**Pull request by jsanemet**
Additionally:
- Fixed a few things on the 'main' function to get the test cases to pass.
- Export coverage as html for review.
- Indicate which directories should flake8 only transverse in order to avoid it from checking ignored files.
- Ignore PyCharm's cache

### CODE REVIEW

**Review by cescgina — APPROVED** [review]
Looks great, good job!

**Review by bregman-arie — APPROVED** [review]
Looks really good

File: tox.ini [hunk 1]
```diff
@@ -16,10 +16,7 @@ usedevelop = True
 commands = {posargs}
 
 [testenv:pep8]
-commands = flake8 {posargs}
-
-[flake8]
-exclude=.tox
```
Comment (sdatko) [inline tox.ini:22]: I am not so convinced about this one, I think the `{posargs}` have no real sense in `pep8` venv, but let it be...

**Review by sdatko — APPROVED** [review]
I am 98% happy, which is more than enough for me. Good job!

/edit: The missing 2% IMOs:
– the commit `Merge remote-tracking branch 'origin/main' into browser-unit` should be gone from list of changes...
– this entries in `tox.ini` are bothering me now...

**Pull request merged**